app/models/registration_system.rb

```diff
@@ -83,8 +83,9 @@ def self.update_rhsm_conf(options = {})
 
     return unless option_values[:proxy_address]
 
-    write_rhsm_config(:proxy_hostname => option_values[:proxy_address].split(':')[0],
-                      :proxy_port     => option_values[:proxy_address].split(':')[1],
+    proxy_uri = URI.parse(option_values[:proxy_address].include?("://") ? option_values[:proxy_address] : "http://#{option_values[:proxy_address]}")
```

Comment:
I feel like this just punts the issue down the road. Is there no field validation for this? Couldn't we just say "This thing needs to be a valid URI" and give a flash message otherwise?

For example if someone were forget a `/` and enter "http:/example.com:123" you would end up in this situation right?

```
irb(main):007:0> uri = URI.parse("http://http:/example.com:123")
=> #<URI::HTTP http://http/example.com:123>
irb(main):008:0> uri.host
=> "http"
irb(main):009:0> uri.port
=> 80
```

Reply:
I agree - I tried using `Addressable::URI` objects initially before I did string checking but am not sure there is error checking by the caller of this method, and testing with that manner threw other ugly exceptions further down the stack that similar didn't appear in the UI or any easy-to-interpret log, so I instead just altered the string validation.

I can look at it deeper after this week, I am onsite with a customer now unfortunately, or we can close this PR in favor of a method that does better error handling.

Reply:
¯\\_(ツ)_/¯ I think I'm okay with this as is for now.  @carbonin and I were just discussing that we may want to open a RFE to the UI to add validation to this field.  At least this adds the ability to accept URLs